ios-target/Model/Target.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Target.swift
+//  ios-target
+//
+//  Created by sol manini on 6/12/19.
+//  Copyright © 2019 TopTier labs. All rights reserved.
+//
+
+import Foundation
+
+class Target: Codable {
+  var id: Int
+  var title: String
+  var lat: Double
+  var lng: Double
+  var radius: Double
+  var topicId: Int
+  
+  private enum CodingKeys: String, CodingKey {
+    case id
+    case title
+    case lat
+    case lng
+    case radius
+    case topicId = "topic_id"
+  }
+  
+  init(id: Int, title: String = "", lat: Double, lng: Double = 0,
+       radius: Double = 0, topicId: Int = 0) {
+    self.id = id
+    self.title = title
+    self.lat = lat
+    self.lng = lng
+    self.radius = radius
+    self.topicId = topicId
+  }
+  
+  //MARK: Codable
+  
+  func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode(id, forKey: .id)
+    try container.encode(title, forKey: .title)
+    try container.encode(lat, forKey: .lat)
+    try container.encode(lng, forKey: .lng)
+    try container.encode(radius, forKey: .radius)
+    try container.encode(topicId, forKey: .topicId)
+  }
+  
+  required init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    do {
+      id = try container.decode(Int.self, forKey: .id)
+    } catch {
+      id = Int(try container.decode(Int.self, forKey: .id))
```

Comment:
if the line above throws an exception this line will do the same. What exactly you want to achieve here?